src/acl_kernel.cpp

```diff
@@ -2785,7 +2785,14 @@ l_copy_and_adjust_arguments_for_device(cl_kernel kernel, cl_device_id device,
     // Exclude kernel argument value from device-side buffer by default.
     cl_uint buf_incr = 0;
 
-    if (arg_info->addr_space == ACL_ARG_ADDR_LOCAL) {
+    if (arg_info->streaming_info) {
+#ifdef MEM_DEBUG_MSG
+      printf("streaming");
+#endif
+      stream_info.args.emplace_back(aocl_mmd_streaming_kernel_arg_info_t{
+          arg_info->streaming_info->interface_name.c_str(),
+          &kernel->arg_value[host_idx], arg_info->size});
```

Comment:
This is unsafe, since the argument value may change before invocation:
https://github.com/intel/fpga-runtime-for-opencl/blob/bd7f1073b7a85858b5180a3329a01286f34bc8d8/include/acl_types.h#L407-L417